Add package publishing step

julienduchesne opened the pull request:
Issue: https://github.com/grafana/deployment_tools/issues/36289
Based on the new image: https://github.com/grafana/deployment_tools/tree/master/docker/package-publish
This is a new step meant to replace the store-packages command. It will greatly improve publishing performace and it publishes to a common repository shared with all Grafana products

Failures are set to be ignored temporarily. So that it can be tested and switched over when we know it works!

julienduchesne commented:
I had forgotten to place it ready for review 🤦 🤦 🤦 . No wonder it wasn't getting attention...

dsotirakis reviewed (approved):
Thanks a lot @julienduchesne 🙇 LGTM!

dsotirakis committed:
Add package publishing step
Issue: https://github.com/grafana/deployment_tools/issues/36289
Based on the new image: https://github.com/grafana/deployment_tools/tree/master/docker/package-publish
This is a new step meant to replace the store-packages command. It will greatly improve publishing performace and it publishes to a common repository shared with all Grafana products

dsotirakis commented:
Rebased and now shows that the commits are mine, but they are not! 
Going to merge it 💪
